obspy/io/mseed/core.py

```diff
@@ -785,6 +797,17 @@ def _write_mseed(stream, filename, encoding=None, reclen=None, byteorder=None,
                 trace_attr['encoding'] = 1
             elif trace.data.dtype.type == np.dtype(native_str('|S1')).type:
                 trace_attr['encoding'] = 0
+            # int64 data not supported; if possible downcast to int32, else create error message
+            # After bumping up to numpy 1.9.0 this check can be replaced by numpy.can_cast()
+            elif trace.data.dtype.type == np.int64:
+                # check if data can be safely downcast to int32
+                trace_data1 = _np_copy_astype(trace.data, np.int32)
+                if np.array_equal(trace.data, trace_data1):
```

Comment:
How about just checking if the values in the array are in bounds with `np.iinfo(np.int32)` before casting instead of doing the cast + equality check? Feels a bit cleaner and should also be faster.

Reply:
True, that does look smoother, I'll change to that!